.github/actions/buildkite/action.yml

```
@@ -0,0 +1,64 @@
+---
+name: Buildkite Pipeline GitHub Action
+description: A GitHub Action for triggering a build on a Buildkite pipeline.
+inputs:
+  vaultUrl:
+    description: 'Vault URL'
+    required: true
+  vaultRoleId:
+    description: 'Vault role ID'
+    required: true
+  vaultSecretId:
+    description: 'Vault secret ID'
+    required: true
+  secret:
+    description: 'Vault secret with the token field.'
+    default: secret/observability-team/ci/buildkite-automation
+    required: false
+  org:
+    description: 'BuildKite org to interact with.'
+    default: elastic
+    required: false
+  pipeline:
+    description: 'BuildKite pipeline to interact with.'
+    required: true
+  waitFor:
+    description: 'Whether to wait for the build to finish.'
+    default: false
+    required: false
+  printBuildLogs:
+    description: 'Whether to wait for the build to finish.'
+    default: false
+    required: false
+  buildEnvVars:
+    description: 'Additional environment variables to set on the build.'
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+      - uses: hashicorp/vault-action@v2.4.2
+        with:
+          url: ${{ inputs.vaultUrl }}
+          roleId: ${{ inputs.vaultRoleId }}
+          secretId: ${{ inputs.vaultSecretId }}
+          method: approle
+          secrets: |
+            ${{ inputs.secret }} token | BUILDKITE_API_ACCESS_TOKEN
+
+      - run: echo "${{ github.action_path }}" >> $GITHUB_PATH
+        shell: bash
+
+      - name: Trigger Buildkite pipeline
+        run: |
+          run.sh \
```

I forgot to mention that you can use the full path directly and make it in one step only

```suggestion
          ${{ github.action_path }}/run.sh \
```

---

https://github.com/elastic/apm-pipeline-library/pull/2033/commits/3622e2d14cabfcfa77db91f4accca1b38ead9a2e